src/aws-cpp-sdk-core/source/internal/AWSHttpResourceClient.cpp

```diff
@@ -187,7 +187,8 @@ namespace Aws
             AWSHttpResourceClient(clientConfiguration, EC2_METADATA_CLIENT_LOG_TAG),
             m_endpoint(endpoint),
             m_disableIMDS(clientConfiguration.disableIMDS),
-            m_tokenRequired(true)
+            m_tokenRequired(true),
+            m_disableIMDSV1(clientConfiguration.disableImdsV1)
```

Comment:
you have forgotten to update the regular constructor
(or add a default value)

Reply:
good catch, added default value